testing/integration/test-executor/testapp/config/AppConfig.h.in

```diff
@@ -0,0 +1,66 @@
+/*******************************************************************************
+ * Copyright (c) 2010-2019 Robert Bosch GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *
+ ******************************************************************************/
+
+/**
+ * @brief App config header
+ *
+ * @details
+ * ```
+ *    ______ _____   _   _ _____ _____  ___  ______________ _____________   __
+ *    |  _  \  _  | | \ | |  _  |_   _| |  \/  |  _  |  _  \_   _|  ___\ \ / /
+ *    | | | | | | | |  \| | | | | | |   | .  . | | | | | | | | | | |_   \ V /
+ *    | | | | | | | | . ` | | | | | |   | |\/| | | | | | | | | | |  _|   \ /
+ *    | |/ /\ \_/ / | |\  \ \_/ / | |   | |  | \ \_/ / |/ / _| |_| |     | |
+ *    |___/  \___/  \_| \_/\___/  \_/   \_|  |_/\___/|___/  \___/\_|     \_/
+ *
+ *
+ *     _____ _   _ _____ _____   _   _  _____  ___ ______ ___________ _
+ *    |_   _| | | |_   _/  ___| | | | ||  ___|/ _ \|  _  \  ___| ___ \ |
+ *      | | | |_| | | | \ `--.  | |_| || |__ / /_\ \ | | | |__ | |_/ / |
+ *      | | |  _  | | |  `--. \ |  _  ||  __||  _  | | | |  __||    /| |
+ *      | | | | | |_| |_/\__/ / | | | || |___| | | | |/ /| |___| |\ \|_|
+ *      \_/ \_| |_/\___/\____/  \_| |_/\____/\_| |_/___/ \____/\_| \_(_)
+ * ```
+ *
+ * This file is managed through cmake! The <headername>.h.in file acts as the
+ * template for generating the "real" header during cmakes' configure stage.
+ * This generated header will be placed in your CMAKE_CURRENT_BINARY_DIR and
+ * included during building.
+ *
+ * In the template we refer to cmake variables through ${XYZ} or @XYZ@ notation.
+ * Any such occurrence will be populated by cmake with the coresponding variable
+ * value.
+ *
+ * The intended way of modifying values in this header is:
+ *  A) Through the <headername>.cmake file located either in either the boards'
+ *     or the apps' config folder.
+ *  B) Through command line options such as:
+ *     `$ cmake --build <BUILD_DIR> -D <VAR>=<VAL>`
+ *     Where <VAR> is the name of a cmake variable and <VAL> it's associated
+ *     value.
```

Comment:
I don't think this is the intended way anymore. We reverted back to doing configuration directly in the headers and then letting cmake copy and replace them as needed.